### PR TITLE
Blog post: Cloudflare Slack bot

### LINKED ISSUE #782: Blog Post: How to create a Cloudflare Workers - Including scheduled/trigger

### PR DESCRIPTION
**Pull request by TuvalSimha**
Closes #782

### CODE REVIEW

File: pages/blog/slack-bot_with_cloudflare.mdx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+// We use Octokit to use GitHub API
+... WRITE ANY CODE YOU WANT HERE
+}
+```
+
+### How can I schedule my function in the Cloudflare Worker?
+
+After you finish the installation, Cloudflare creates also `tmol` file, this file contains the name, main, compatibility_date and triggers. To use the triggers, Cloudflare supports Crons and is look like that:
+
+```tmol
```
Comment (B2o5T) [inline pages/blog/slack-bot_with_cloudflare.mdx:89]:
```suggestion
```toml
```

File: pages/blog/slack-bot_with_cloudflare.mdx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+ // We use Slack Client to send messages
+  const octokit = new Octokit({ auth: env.GH_BOT_TOKEN });
+// We use Octokit to use GitHub API
+... WRITE ANY CODE YOU WANT HERE
+}
+```
+
+### How can I schedule my function in the Cloudflare Worker?
+
+After you finish the installation, Cloudflare creates also `tmol` file, this file contains the name, main, compatibility_date and triggers. To use the triggers, Cloudflare supports Crons and is look like that:
```
Comment (B2o5T) [inline pages/blog/slack-bot_with_cloudflare.mdx:87]:
```suggestion
After you finish the installation, Cloudflare creates also `toml` file, this file contains the name, main, compatibility_date and triggers. To use the triggers, Cloudflare supports Crons and is look like that:
```
toml I guess?

File: pages/blog/slack-bot_with_cloudflare.mdx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        Authorization: Bearer ${GH_BOT_TOKEN}
+documents: 'src/**/*.ts'
+generates:
+  types.ts:
+    config:
+      scalars:
+        URI: string
+    plugins:
+      - 'typescript'
+      - 'typescript-operations'
```
Comment (B2o5T) [inline pages/blog/slack-bot_with_cloudflare.mdx:131]:
```suggestion
      - typescript
      - typescript-operations
```

File: pages/blog/slack-bot_with_cloudflare.mdx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+#### Why did I use GraphQL Codegen?
+
+- GraphQL Codegen generates types-based schema - The types are fully type-safe.
+- The process to add GraphQL Codegen to the project is easy, fast and reduces type error.
+
+## What else you can do with Cloudflare Workers? (Hint? Yoga)
+
+**GraphQL Yoga,** one of the greatest tools in The Guild, provides a GraphQL server. Our focus is on easy setup, performance and great developer experience. So you can easily integrate it into any platform besides Node.js.
+
+You can find more information and tutorials on how to integrate **Cloudflare Workers** with **GrpahQL Yoga - [LINK](https://www.graphql-yoga.com/docs/integrations/integration-with-cloudflare-workers)**
```
Comment (B2o5T) [inline pages/blog/slack-bot_with_cloudflare.mdx:145]:
```suggestion
You can find more information and tutorials on how to integrate **Cloudflare Workers** with **GrpahQL Yoga - [link](https://www.graphql-yoga.com/docs/integrations/integration-with-cloudflare-workers)**
```

File: pages/blog/slack-bot_with_cloudflare.mdx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+Here you can see the `run` function, we can write any code we want to schedule in the Cloudflare Workers. We create a Slack bot that sends a daily message about all open pull requests.
+
+```ts
+export async function run(env: Env) {
+  const slack = createSlackClient(env.SLACK_TOKEN);
+ // We use Slack Client to send messages
+  const octokit = new Octokit({ auth: env.GH_BOT_TOKEN });
+// We use Octokit to use GitHub API
+... WRITE ANY CODE YOU WANT HERE
```
Comment (B2o5T) [inline pages/blog/slack-bot_with_cloudflare.mdx:81]:
```suggestion
  // We use Octokit to use GitHub API
  ... WRITE ANY CODE YOU WANT HERE
```

File: pages/blog/slack-bot_with_cloudflare.mdx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+The Workers platform comes with a new set of tools, and super nice APIs to integrate directly into the deployment flow you want to design. At a high level, we can create a lot of cool features:
+
+- Build an entire Slack bot.
+- Return small HTML page.
+- JSON return.
+- Fetch HTML / JSON.
+- A/B Testing.
+- Auth with headers.
+- Schedule processing
+- And a lot more examples… ([You can find a lot of examples in Cloudflare docs](https://developers.Cloudflare.com/workers/examples/)).
```
Comment (B2o5T) [inline pages/blog/slack-bot_with_cloudflare.mdx:37]:
```suggestion
- Build an entire Slack bot
- Return small HTML page
- JSON return
- Fetch HTML / JSON
- A/B Testing
- Auth with headers
- Schedule processing
- And a lot more examples… ([You can find a lot of examples in Cloudflare docs](https://developers.Cloudflare.com/workers/examples/))
```

File: pages/blog/slack-bot_with_cloudflare.mdx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+export default withArticle(meta)
+ 
+# Introduction
+
+Big or small companies are super complex systems, every company needs to find the right way to work efficiently. There is a lot of applications that help us to manage our daily job to the maximum.
+Lately, we found some black holes in The Guild employees-system process.
+
+So we thought about some solutions and decided to create an internal tool that help us, and we thought it might be valuable for others.
+
+So, let's talk about what we created - **GitHub Slack bot reminder in Cloudflare Workers.**
```
Comment (B2o5T) [inline pages/blog/slack-bot_with_cloudflare.mdx:19]:
```suggestion
So, let's talk about what we created - **GitHub Slack bot reminder in Cloudflare Workers**.
```

File: pages/blog/slack-bot_with_cloudflare.mdx [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+Like the example above, we define the schema path, add Authorization to the header and after that, we generate the schema and create types to use in `run` function.
+
+#### Why did I use GraphQL Codegen?
+
+- GraphQL Codegen generates types-based schema - The types are fully type-safe.
+- The process to add GraphQL Codegen to the project is easy, fast and reduces type error.
+
+## What else you can do with Cloudflare Workers? (Hint? Yoga)
+
+**GraphQL Yoga,** one of the greatest tools in The Guild, provides a GraphQL server. Our focus is on easy setup, performance and great developer experience. So you can easily integrate it into any platform besides Node.js.
```
Comment (kamilkisiela) [inline pages/blog/slack-bot_with_cloudflare.mdx:143]:
```suggestion
**GraphQL Yoga**, one of the greatest tools in The Guild, provides a GraphQL server. Our focus is on easy setup, performance and great developer experience. So you can easily integrate it into any platform besides Node.js.
```

File: pages/blog/slack-bot_with_cloudflare.mdx [hunk 9]
```diff
@@ -0,0 +1,145 @@
+import withArticle from '../../ui/blog/article'
+export const meta = {
+  title: 'Building Slack bot with Cloudflare Workers',
+  author: 'tuvalSimha',
+  tags: ['graphql', 'yoga', 'codegen'],
```
Comment (kamilkisiela) [inline pages/blog/slack-bot_with_cloudflare.mdx:5]:
```suggestion
  tags: ['graphql', 'codegen'],
```
The Slack bot doesn't use Yoga. I'm also thinking about a new category for such things, maybe `edge` or `tutorials`? It kind of touches GraphQL and Codegen but the main topic here is CF Worker and its scheduler feature.

File: pages/blog/slack-bot_with_cloudflare.mdx [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+    config:
+      scalars:
+        URI: string
+    plugins:
+      - typescript
+      - typescript-operations
+```
+
+Like the example above, we define the schema path, add Authorization to the header and after that, we generate the schema and create types to use in `run` function.
+
```
Comment (kamilkisiela) [inline pages/blog/slack-bot_with_cloudflare.mdx:135]: Maybe it should include an example call to Github API with `Octokit` + `Codegen` setup?

File: pages/blog/slack-bot-with-cloudflare.mdx [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+    config:
+      scalars:
+        URI: string
+    plugins:
+      - typescript
+      - typescript-operations
+```
+
+Like the example above, we define the schema path, add Authorization to the header and after that, we generate the schema and create types to use in `run` function.
+
```
Comment (charlypoly) [inline pages/blog/slack-bot-with-cloudflare.mdx:135]: putting back @kamilkisiela's feedback:

image

**Review by charlypoly — CHANGES_REQUESTED** [review]
@TuvalSimha, nice article! 🤓 

I've left some suggestions for phrasing, mainly to avoid the use of too many combined superlatives (ex: "very super [...]").

**Comment by charlypoly** [comment]
@TuvalSimha, I also fixed the Vercel build 🤓 

Let me know if you need some illustrations 👨🏼‍🎨

**Review by charlypoly — APPROVED** [review]
Let's publish tomorrow

**Pull request merged**